test/modules/core/core/attribute.spec.js

```diff
@@ -120,3 +120,98 @@ test('WebGL#Attribute getValue', t => {
 
   t.end();
 });
+
+// If the vertex shader has more components than the array provides,
+// the extras are given values from the vector (0, 0, 0, 1) for the missing XYZW components.
+// https://www.khronos.org/opengl/wiki/Vertex_Specification#Vertex_format
+test('Attribute#missing component', t => {
+
+  if (fixture.gl.getExtension('STACKGL_resize_drawingbuffer')) {
+    // headless-gl does not seem to implement this behavior
+    t.comment('Skipping headless-gl');
+    t.end();
+    return;
+  }
+
+  const contexts = {
+    WebGL1: fixture.gl,
+    WebGL2: fixture.gl2
+  };
+
+  const modelProps = {
+    vs: `
+  attribute vec3 position;
+  attribute vec4 color;
+  varying vec4 vColor;
+  void main(void) {
+    vColor = vec4(position.xz, color.ra / 255.);
+    gl_Position = vec4(position.xy, 0.0, 1.0);
+    gl_PointSize = 2.0;
+  }
+  `,
+    fs: `
+  precision highp float;
+  varying vec4 vColor;
+  void main(void) {
+    gl_FragColor = vColor;
+  }
+  `,
+    drawMode: GL.POINTS,
+    vertexCount: 4
+  };
+
+  const testCases = [
+    {
+      position: {
+        size: 2,
+        value: new Float32Array([-1, -1, 1, -1, -1, 1, 1, 1])
+      },
+      color: {
+        size: 3,
+        value: new Uint8ClampedArray([32, 0, 0, 64, 0, 0, 128, 0, 0, 255, 0, 0])
+      }
+    },
+    {
+      position: {
+        size: 2,
+        stride: 12, // 3 * 4 bytes per element
```

Comment:
Testing `stride`, while valuable, seems like a separate thing, as it relates to "manually" skipping over datums?

My assumption was that this test case should focus on the "automatic" matching of different attribute size in JS vs attribute declaration size in GLSL, making sure components are ignored / filled in from (0,0,0,1) as appropriate?

Reply:
I want to make sure that the skipped over element doesn't "bleed" into the attribute (which is the case in headless-gl).

Reply:
> I want to make sure that the skipped over element doesn't "bleed" into the attribute (which is the case in headless-gl).

Interesting. Is this `bleeding` behavior different with/without `stride`? I.e. if no stride, no bleeding?

Also maybe add this comment to the test case?

Reply:
With no stride and a shorter typed array, render silently fails (all pixels blank).